### PR TITLE
Migration to Angular/NX v15

### PR DESCRIPTION
**Pull request by DelBiss**
Migration executed by Nx with the command 'nx migrate 15.9.3', the latest non-major version of v15

Those main dependancies have been updated by the automatic migration:
- **Nx**: from v14.4..2 to v15.9.3
- **Angular**: from 14.0.5 to v15.2.10
- **NgRx**: from 14.0.1 to v15.3.0
- **TypeScript**:from 4.7.4 to 4.9.5

The migration caused some TypeScript error, but also problems with other dependancies that needed to be updated:
- **jest-marble**: from 2.5.1 to 3.0.6
- **rxjs**: from 6.6.0 to 7.8.1

## About the Typescript Issue following migration from 4.7.4 to 4.9.5
Most of the error was due to lack of constrain on certain type.
There was also some weird one in `async-action.trait.ts` and comment have been left to explain them.

The most noteworthy one is related to `KeyedConfig` and `AllTraitConfigs` (because of the TODO comment on line 75.
The change that I've made is the following:
  ```typescript
  - KC = KeyedConfig<KEY, C>
  + KC extends AllTraitConfigs = KeyedConfig<KEY, C>
  ``` 
What I would like to note is that this following change look like to also work:
  ```typescript
  - KC = KeyedConfig<KEY, C>
  + KC extends KeyedConfig<KEY, C>= KeyedConfig<KEY, C>
  ``` 
I've chose to go with `AllTraitConfigs` since it's the least constrain needed to make it work.

## Deprication warning on RxJS
There was 2 deprication warning in `trait-effect.spec`.

The first one concerning `mapTo(val)`. It have been replace by `map(()=>val)` has suggested.

The second one's concerning the usage of `toPromise()` on Observable.
I've tried to implement the [suggested change](https://rxjs.dev/deprecations/to-promise), but it doesn't work.
Maybe it's because I didn't do it properly or their's a bug in the test or effects.
Since the test work without the change and a deprecation warning, nothing have been change for the moment other than commenting.

## Final Note

- This PR containt the commit in PR #37 
- This is my first time contributing to OpenSource, comment on how to make better contribution are welcome.
- This PR pass all the unit test.
- This PR pass the build of the app and librairies
- This PR pass the manual testing of the app.

### CODE REVIEW

**Comment by DelBiss** [comment]
Also, note that I'm currently working on migration to v16

**Comment by gabrielguerrero** [comment]
@DelBiss Thanks a lot for this will check on it and try to merge it soon

File: jest.preset.js [hunk 1]
```diff
@@ -15,4 +15,14 @@ module.exports = {
       lines: 50,
     },
   },
+  /* TODO: Update to latest Jest snapshotFormat
+   * By default Nx has kept the older style of Jest Snapshot formats
+   * to prevent breaking of any existing tests with snapshots.
+   * It's recommend you update to the latest format.
+   * You can do this by removing snapshotFormat property
+   * and running tests with --update-snapshot flag.
+   * Example: "nx affected --targets=test --update-snapshot"
```
Comment (gabrielguerrero) [inline jest.preset.js:24]: there is no snapshot testing in the repo so I think we can safestly update to the latest format

**Comment by gabrielguerrero** [comment]
@DelBiss the PR looks good just some small comments, I will like to merge this let me know if you have time to do the changes I requested Or I can do them, I saw some where you were also trying to migrate to 16 let me know if you made any progress on that, otherwise I can do it, I want to migrate to 17 there is a new package Im building that have traits that work with ngrx-signals that needs the version 17

**Pull request merged**